src/components/Settings/SettingsMain.tsx

```diff
@@ -116,76 +142,96 @@ const SettingsMain: React.FC = () => {
           intl.formatMessage(globalMessages.settings),
         ]}
       />
-      <div className="mb-6">
-        <h3 className="heading">
-          {intl.formatMessage(messages.generalsettings)}
-        </h3>
-        <p className="description">
-          {intl.formatMessage(messages.generalsettingsDescription)}
-        </p>
-      </div>
-      <div className="section">
-        <Formik
-          initialValues={{
-            applicationTitle: data?.applicationTitle,
-            applicationUrl: data?.applicationUrl,
-            csrfProtection: data?.csrfProtection,
-            hideAvailable: data?.hideAvailable,
-            locale: data?.locale ?? 'en',
-            region: data?.region,
-            originalLanguage: data?.originalLanguage,
-            partialRequestsEnabled: data?.partialRequestsEnabled,
-            trustProxy: data?.trustProxy,
-          }}
-          enableReinitialize
-          validationSchema={MainSettingsSchema}
-          onSubmit={async (values) => {
-            try {
-              await axios.post('/api/v1/settings/main', {
-                applicationTitle: values.applicationTitle,
-                applicationUrl: values.applicationUrl,
-                csrfProtection: values.csrfProtection,
-                hideAvailable: values.hideAvailable,
-                locale: values.locale,
-                region: values.region,
-                originalLanguage: values.originalLanguage,
-                partialRequestsEnabled: values.partialRequestsEnabled,
-                trustProxy: values.trustProxy,
-              });
-              mutate('/api/v1/settings/public');
-
-              if (setLocale) {
-                setLocale(
-                  (userData?.locale
-                    ? userData.locale
-                    : values.locale) as AvailableLocale
-                );
-              }
+      <Formik
+        initialValues={{
+          applicationTitle: data?.applicationTitle,
+          applicationUrl: data?.applicationUrl,
+          csrfProtection: data?.csrfProtection,
+          hideAvailable: data?.hideAvailable,
+          locale: data?.locale ?? 'en',
+          region: data?.region,
+          originalLanguage: data?.originalLanguage,
+          partialRequestsEnabled: data?.partialRequestsEnabled,
+          trustProxy: data?.trustProxy,
+          proxyEnabled: data?.proxy.enabled,
+          proxyHostname: data?.proxy.options.hostname,
+          proxyPort: data?.proxy.options.port,
+          proxyUseSSL: data?.proxy.options.useSSL,
+          proxyAuthUser: data?.proxy.options.authUser,
+          proxyAuthPass: data?.proxy.options.authPass,
+          proxyBypassLocalAddresses: data?.proxy.options.bypassLocalAddresses,
+          proxyIgnoredAddresses: data?.proxy.options.ignoredAddresses ?? [],
```

Comment:
Similar to the Tautulli settings not being a part of Plex settings, I think this can have its own settings endpoint especially since you have it in its own section.  There's too much here.